File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/das/DasBaseInfoParser.java
package com.shmedo.configlibrary.iot.cmd.parser.das;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;
import com.shmedo.configlibrary.iot.model.das.DasBaseInfo;

import java.util.HashMap;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2021/4/16 <br/>
 * 描述：      解析DAS 状态页面基本信息参数
 */
public class DasBaseInfoParser implements IOTResultParser<DasBaseInfo> {
    @Override
    public DasBaseInfo parse(String result) {
        DasBaseInfo info = new DasBaseInfo();
        try {
            String[] keyValues = result.split("&");
            HashMap<String, String> keyValueMap = new HashMap<>();

            for (String keyValue : keyValues) {
                String[] strs = keyValue.split("=");
                if (strs.length < 2) {
                    keyValueMap.put(strs[0], "");
                } else {
                    keyValueMap.put(strs[0], strs[1]);
                }
            }
            info.setSn(keyValueMap.getOrDefault("sn", "NullKey"));
            info.setIccid(keyValueMap.getOrDefault("iccid", "NullKey"));
            info.setImei(keyValueMap.getOrDefault("imei", "NullKey"));
            info.setVer(keyValueMap.getOrDefault("ver", "NullKey"));
            info.setLocal(keyValueMap.getOrDefault("local", "NullKey"));
            info.setInvolt(keyValueMap.getOrDefault("involt", "NullKey"));
            info.setOutvolt(keyValueMap.getOrDefault("outvolt", "NullKey"));
            info.setCsq(keyValueMap.getOrDefault("csq", "NullKey"));
            info.setIsp(keyValueMap.getOrDefault("isp", "NullKey"));
            info.setCode(keyValueMap.getOrDefault("code", "NullKey"));

            return info;
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.DAS_MD_GET_DEVICE_BASE;
    }
}

File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/das/DasBaseInfoParser.kt
package com.shmedo.configlibrary.iot.cmd.parser.das

import com.shmedo.configlibrary.iot.enums.IOTCommandType
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser
import com.shmedo.configlibrary.iot.model.das.DasBaseInfo

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2021/4/16 <br></br>
 * 描述：      解析DAS 状态页面基本信息参数
 */
class DasBaseInfoParser : IOTResultParser<DasBaseInfo?> {
    override fun parse(result: String): DasBaseInfo? {
        val info = DasBaseInfo()
        return try {
            val keyValues = result.split("&").toTypedArray()
            val keyValueMap = HashMap<String, String>()
            for (keyValue in keyValues) {
                val strs = keyValue.split("=").toTypedArray()
                if (strs.size < 2) {
                    keyValueMap[strs[0]] = ""
                } else {
                    keyValueMap[strs[0]] = strs[1]
                }
            }
            info.sn = keyValueMap.getOrDefault("sn", "NullKey")
            info.iccid = keyValueMap.getOrDefault("iccid", "NullKey")
            info.imei = keyValueMap.getOrDefault("imei", "NullKey")
            info.ver = keyValueMap.getOrDefault("ver", "NullKey")
            info.local = keyValueMap.getOrDefault("local", "NullKey")
            info.involt = keyValueMap.getOrDefault("involt", "NullKey")
            info.outvolt = keyValueMap.getOrDefault("outvolt", "NullKey")
            info.csq = keyValueMap.getOrDefault("csq", "NullKey")
            info.isp = keyValueMap.getOrDefault("isp", "NullKey")
            info.code = keyValueMap.getOrDefault("code", "NullKey")
            info
        } catch (ex: Exception) {
            ex.printStackTrace()
            null
        }
    }

    override fun validate(result: String) {}
    override fun commandType(): IOTCommandType {
        return IOTCommandType.DAS_MD_GET_DEVICE_BASE
    }
}